### ISSUE TITLE
[BUG] - EzSNMPError: USM unknown security name (no such user exists)

### ISSUE THREAD

**Issue by carlkidcrypto** [carlkidcrypto/ezsnmp#50]
This issue was observed in #44. When running under a multi-thread or multi-process application
V3 session will show
`EzSNMPError: USM unknown security name (no such user exists)` errors.
This doesn't appear consistent. It appears random at this moment.

@mtuska opening this new issue to track the USM error.

**Comment by carlkidcrypto** [comment]
Update: I can now turn the problem on/off.

This is the problem code:

```c++
void __remove_user_from_cache(struct session_list *ss)
{
    struct usmUser *actUser = usm_get_userList();
    while (actUser != NULL)
    {
        struct usmUser *dummy = actUser;
        if (actUser->secName != NULL && ss->session->securityName != NULL &&
            actUser->engineID != NULL && ss->session->contextEngineID != NULL &&
            strcmp((const char *)dummy->secName, (const char *)ss->session->securityName) == 0 &&
            strcmp((const char *)dummy->engineID, (const char *)ss->session->contextEngineID) == 0)
        {
            usm_remove_user(actUser);
            actUser->next = NULL;
            actUser->prev = NULL;
            usm_free_user(actUser);
            break;
        }
        actUser = dummy->next;
    }
}```

**Comment by carlkidcrypto** [comment]
@mtuska go ahead and try the proposed fix in #108 . Let me know if it makes a difference.
